### PR TITLE
kubelogin-oidc: init at 1.23.0

### PR DESCRIPTION
**Pull request by benley**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118694` at 49e09bd6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>kubelogin-azure</li>
    <li>kubelogin-oidc</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23830,7 +23830,11 @@ in
 
   kubeless = callPackage ../applications/networking/cluster/kubeless { };
 
-  kubelogin = callPackage ../applications/networking/cluster/kubelogin { };
+  kubelogin = kubelogin-azure;
+
+  kubelogin-azure = callPackage ../applications/networking/cluster/kubelogin { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:23835]: Please put this into a separate commit.

File: pkgs/applications/networking/cluster/kubelogin-oidc/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "v${version}";
+    sha256 = "0n94nx17c6ln2nd6d9yr93vc251y1xphq1wj2vzs4j2l8dqfyjpn";
+  };
+
+  subPackages = ["."];
+
+  vendorSha256 = "1dvrk6z6k66wawgb50n8hbgdd8fly399mlbgnvxi671vfi7lkz09";
+
+  postInstall = ''
+    ln -s $out/bin/kubelogin $out/bin/kubectl-oidc_login
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/kubelogin-oidc/default.nix:19]: Does this conflict with the other kubelogin?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/kubelogin-oidc/default.nix (hunk 2 above)
Comment (benley) [reply pkgs/applications/networking/cluster/kubelogin-oidc/default.nix:19]: Probably! Both apps ship a binary called `kubelogin`.  It's somewhat unlikely that a user will need to install both simultaneously, but if that is a concern perhaps we could rename this one entirely instead of adding the symlink?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (benley) [reply pkgs/top-level/all-packages.nix]: done.
